api/opencl_architecture.asciidoc

```diff
@@ -312,7 +312,7 @@ Each work-items global ID is an N-dimensional tuple.
 The global ID components are values in the range from F, to F plus the
 number of elements in that dimension minus one.
 
-If a kernel is created from OpenCL 2.0 or SPIR-V, the size of work-groups in
+If a kernel is created from OpenCL C 2.0 or SPIR-V, the size of work-groups in
```

Comment:
This sentence is more correct now than it was before, but it still feels incomplete.  Presumably kernels created from implementation-defined binary objects may also support non-uniform work groups, at least in some cases?

Reply:
Yes, that makes sense.  Likewise I see no reason why a builtin kernel couldn't support non-uniform work groups.

I've flipped the wording to

> Unless a kernel comes from a source that disallows it, e.g. OpenCL C 1.x or using `-cl-uniform-work-group-size`, ...